bofire/data_models/strategies/predictives/botorch.py

```diff
@@ -137,6 +317,18 @@ def is_constraint_implemented(cls, my_type: Type[Constraint]) -> bool:
             return False
         return True
 
+    @model_validator(mode="after")
+    def validate_exclusive_local_search_trust_region(self):
+        # dont allow both local search and trust region to be used at the same time
+        if (
+            self.local_search_config is not None
+            and self.trust_region_config is not None
+        ):
+            raise ValueError(
+                "Local search and trust region optimization cannot be used at the same time."
+            )
+        return self
```

Comment:
If we do it as we do it currently, we should definitly validate that we only have one output feature if we are using TURBO.

Reply:
Based on my understanding TURBO only works with a GP surrogate with a lengthscale based kernel, so we should check the surrogate and its kernel.

Reply:
Yes so the reference method does indeed scale the inputs to a hypercube and then warp the trust region based on the normalized lengthscales of the GP. From my reading I think the scaling to a hypercube at the start in important but the warping of the box is optional (it's not used in morbo for instance).

The issue is that I am not sure how to add these things here with the minimal footprint. It's in general the challenge of why things like TurBO and MORBO aren't in Ax

Reply:
The turbo validation is moved to the `TURBOConfig` (see comment above), we could then also add to the Turboconfig, if we want to have the warping of the box or not. Based on this hyperparam we could then check if the chosen surrogate is suitable. What do you think?